components/script/dom/window.rs

```diff
@@ -347,6 +347,11 @@ impl Window {
 
     pub fn maybe_window_proxy(&self) -> Option<Root<WindowProxy>> {
```

Comment:
Bikeshed: rename this to `undiscarded_window_proxy`? There may be some places where the spec will require us to use a window proxy, even if it has been discarded. Might also be worth putting a spec link in to explain what "discarded" means here.